client/src/tests/Carousel.test.tsx

```diff
@@ -0,0 +1,111 @@
+import { render, screen, cleanup, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import Carousel from '../components/Carousel';
+import componentSmokeTest from './componentSmokeTest';
+
+afterEach(cleanup);
+
+componentSmokeTest({
+  Component: Carousel,
+  props: {
+    imagePaths: [
+      '/general-overview-ex.png',
+      '/key-concepts-ex.png',
+      '/conclusion-ex.png',
+    ],
+    titles: [
+      'General Overview:',
+      'Key Concepts:',
+      'Conclusion:',
+    ],
+    descriptions: [
+      'A top-level breakdown that captures a succinct summary of the uploaded file contents.',
+      'Highlight the top concepts present with a brief summary.',
+      'Finish off your notes with a comprehensive, succinct conclusion.',
+    ],
+  },
+});
+
+const carousel =
+  <Carousel
+    imagePaths={[
+      '/general-overview-ex.png',
+      '/key-concepts-ex.png',
+      '/conclusion-ex.png',
+    ]}
+    titles={[
+      'General Overview:',
+      'Key Concepts:',
+      'Conclusion:',
+    ]}
+    descriptions={[
+      'A top-level breakdown that captures a succinct summary of the uploaded file contents.',
+      'Highlight the top concepts present with a brief summary.',
+      'Finish off your notes with a comprehensive, succinct conclusion.',
+    ]}
+  />;
+
+function testImage(path: string, button?: number) {
+  render(carousel);
+  if (button != null) {
+    const buttons = screen.getAllByRole('button');
+    fireEvent.click(buttons[button]);
```

Comment:
Is there a way to check that each button has been clicked? Otherwise I'm not sure what the point of this is since I don't think this is verifying that it did anything

Reply:
Oh so the if branch doesn't check anything, it just triggers the left/right button to be clicked (if the caller wants to click left or right of the carousel) so then we can test outside the branch that the page renders the right img when click left/right. I have the if branch so the function will also work when we just want to test the default rendering when not clicking left/right (to refactor code).